tests/pass-dep/libc/libc-epoll.rs

```diff
@@ -269,3 +271,23 @@ fn test_epoll_eventfd() {
     let expected_value = u64::try_from(fd).unwrap();
     assert!(check_epoll_wait::<8>(epfd, vec![(expected_event, expected_value)]));
 }
+
+fn test_pointer() {
+    // Create an epoll instance.
+    let epfd = unsafe { libc::epoll_create1(0) };
+    assert_ne!(epfd, -1);
+
+    // Create a socketpair instance.
+    let mut fds = [-1, -1];
+    let res = unsafe { libc::socketpair(libc::AF_UNIX, libc::SOCK_STREAM, 0, fds.as_mut_ptr()) };
+    assert_eq!(res, 0);
+
+    // Register fd[1] with EPOLLIN|EPOLLOUT|EPOLLET
+    // EPOLLET is negative number for i32 so casting is needed to do proper bitwise OR for u32.
+    let epollet = libc::EPOLLET as u32;
+    let flags = u32::try_from(libc::EPOLLIN | libc::EPOLLOUT | libc::EPOLLRDHUP).unwrap() | epollet;
+    let data = MaybeUninit::<u64>::uninit().as_ptr();
+    let mut ev = libc::epoll_event { events: u32::try_from(flags).unwrap(), u64: data as u64 };
+    let res = unsafe { libc::epoll_ctl(epfd, libc::EPOLL_CTL_ADD, fds[1], &mut ev) };
+    assert_ne!(res, -1);
+}
```

Comment:
@oli-obk Passing pointer to the ``epoll_event``'s field didn't throw any error under the current implementation. But we talked about rejecting pointers before [here](https://hackmd.io/iebj5ey4TzWeHf7VXtSa2Q?comment=af36dc48-05ae-4310-be58-ef32e938c889&utm_source=comment-card&utm_medium=icon). Is the behaviour of this passing test expected?

Reply:
The cast causes an integer address to be produced. I think this means it can't be turned back into an actual pointer, but that's ok since we don't have a use case for it yet.

I think `libc`'s `epoll_event` type is wrong and should be using a union for the data field. I guess for now anyone using this could be using https://doc.rust-lang.org/std/ptr/fn.with_exposed_provenance.html to get back a pointer on some platforms.

We'll need to evaluate whether tokio needs to store pointers here, but we can delay that to later imo

Reply:
Without knowing the context here -- if epoll has to store something that is "either an integer or a pointer", then `Scalar` would be the type to use, I think?